src/GafferUIModule/PathListingWidgetBinding.cpp

```diff
@@ -409,6 +414,15 @@ class PathModel : public QAbstractItemModel
 				c->changedSignal().connect( boost::bind( &PathModel::columnChanged, this ) );
 			}
 
+			IECore::PathMatcher rowSelection;
+			if( m_selection.size() > 0 )
+			{
+				rowSelection = IECore::PathMatcher( m_selection[0] );
+			}
+
+			setSelection( Selection( columns.size(), IECore::PathMatcher() ) );
+			m_selection[0] = rowSelection;
```

Comment:
Yes, that's a logic error on my part. The motivation is from the Light Editor, where switching columns could be common and I think it would be desirable to keep the selected lights. I still think that's worth keeping, especially for the Light Editor since the first column also controls scene selection.

I could be persuaded otherwise, perhaps there are other factors I'm not considering.

Reply:
I like the LightEditor behaviour - I think perhaps this just isn't the right place to implement it.

For Row/Rows mode, it seems reasonable that PathListingWidget should automatically maintain selection in `setColumns()`, because nothing about changing columns changes the interpretation of the selection. So we can just copy `m_selection[0]` to all columns.

But for Cell/Cells mode I don't think PathListingWidget can reason about the new columns at all, so clearing selection makes sense. Unless...maybe we should detect when an old column is reused in `setColumns()` and maintain the selection for any of those? I think this would actually give us the behaviour we want in the LightEditor and doesn't rely on PathListingWidget having any external knowledge of anything.

Reply:
I pushed a new commit fa80d567e6c01842f2d280f6e71384a4055ef096 which I think accomplishes this correctly. `LightEditorInspectorColumn` didn't seem to have a `cellData` binding, so I added one in that commit.

Reply:
Yep, I don't think `PathModel` has any choice but to compare by pointer identity, because Column doesn't have an equality operator. For clients, the rule will be "if you want to maintain selection, reuse existing columns, _don't_ make new columns that happen to be the same". I think that's pretty reasonable...